app/src/main/java/com/example/drawingactivity/MainActivity.kt

```diff
@@ -99,4 +110,85 @@ class MainActivity : AppCompatActivity() {
     }
+
+    private fun editLayerDialog(oldLayerName: String) {
+        viewModel.isEditDialogOpen = true
+        val inflater = LayoutInflater.from(this@MainActivity)
+        val view = inflater.inflate(R.layout.add_layer_dialog, null)
+        val dialog = AlertDialog.Builder(this).apply {
+            setTitle(context.getString(R.string.editLayerDialogTitle))
+            setView(view)
+            val errorView = view.findViewById<TextView>(R.id.errorText)
+            val layerNameInput = view.findViewById<EditText>(R.id.layerNameField)
+            while (!viewModel.defaultLayerHintFound) {
+                val currentHint = getString(R.string.addLayerDefaultName, viewModel.layerNumber)
+                viewModel.createLayerHint(currentHint)
+            }
+            layerNameInput.hint = getString(R.string.addLayerDefaultName, viewModel.layerNumber)
+            if (viewModel.isEditDialogOpen && viewModel.newName != null) {
+                layerNameInput.setText(viewModel.newName)
+                errorView.isVisible = layerNameInput.text.toString().length > NAME_LENGTH_MAX
+            }
+            layerNameInput.addTextChangedListener {
+                viewModel.newName = it.toString().trim()
+                errorView.isVisible = layerNameInput.text.toString().trim().length > NAME_LENGTH_MAX
+            }
+            setPositiveButton(getString(R.string.addLayerPositiveButton)) { _, _ ->
+                viewModel.oldName = oldLayerName
+                var layerName: String = layerNameInput.text.toString().trim()
+                when (layerName) {
+                    getString(R.string.addLayerDefaultName, viewModel.layerNumber) -> {
+                        viewModel.editLayer(layerName)
+                    }
+                    NAME_EMPTY -> {
+                        layerName = getString(R.string.addLayerDefaultName, viewModel.layerNumber)
+                        viewModel.editLayer(layerName)
+                    }
+                    else -> {
+                        viewModel.editLayer(layerName)
+                    }
+                }
+                viewModel.dialogCompleted()
+                openLayerListBottomSheet()
+            }
+            setNegativeButton(getString(R.string.addLayerNegativeButton)) { _, _ ->
+                viewModel.dialogCompleted()
+            }
+            setOnCancelListener {
+                viewModel.dialogCompleted()
+            }
+        }
+        dialog.show()
+    }
+
+    private fun openLayerListBottomSheet() {
+        viewModel.isEditBottomSheetOpen = true
+        if (viewModel.layerList.size == 0) {
+            return
+        }
+        val sheet = BottomSheetDialog(this)
+        val inflater = LayoutInflater.from(this@MainActivity)
+        val view = inflater.inflate(R.layout.recycler_layer_list_layout, null)
+        val data = mutableListOf<AdapterViewModel>()
+        data.add(CloseViewModel(closeSheet = { sheet.dismiss() }))
```

Comment:
if you want there to be a close option in the recyclerview wouldn't it make more sense to be at the top?